netx/dialer/quic_dialer.go

```diff
@@ -0,0 +1,77 @@
+package dialer
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"strconv"
+
+	"github.com/lucas-clemente/quic-go"
+	"github.com/ooni/probe-engine/legacy/netx/dialid"
+	"github.com/ooni/probe-engine/netx/errorx"
+	"github.com/ooni/probe-engine/netx/trace"
+)
+
+// QUICBaseDialer is a system dialer for Quic
+type QUICBaseDialer interface {
+	DialEarlyContext(context.Context, net.PacketConn, net.Addr, string, *tls.Config, *quic.Config) (quic.EarlySession, error)
+}
+
+// QUICContextDialer is a dialer for QUIC using Context.
+type QUICContextDialer interface {
+	DialContext(ctx context.Context, network, addr string, host string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error)
+}
+
+// QUICDialer is the definition of dialer for QUIC assumed by this package.
+type QUICDialer interface {
+	Dial(network, addr string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error)
+}
+
+// QUICSystemDialer is a base dialer for QUIC
+type QUICSystemDialer struct {
+	Saver *trace.Saver
```

Comment:
I see you have a `QUICSaverDialer` below. So maybe it's possible to not have a saver here? It also feels more today this way, probably?

Reply:
Ah, _maybe_ now I understand. This is the saver for the UDP connection while the other is a saver for dialing and for performing the QUIC handshake. I guess I'm a bit confused because I would have expected the saver dialer to take care of all these use cases, rather than having a bit of functionality here. Provided that I have understood correctly, I think it would be better to move this code into quicsaver.go. Otherwise, let's have a chat so that I can better understand what is the purpose of having Saver here!